src/Symfony/Component/Mailer/Bridge/Postal/Transport/PostalTransportFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Postal\Transport;
+
+use Symfony\Component\Mailer\Exception\UnsupportedSchemeException;
+use Symfony\Component\Mailer\Transport\AbstractTransportFactory;
+use Symfony\Component\Mailer\Transport\Dsn;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+
+final class PostalTransportFactory extends AbstractTransportFactory
+{
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+
+        if (!\in_array($scheme, ['postal+api', 'postal'], true)) {
+            throw new UnsupportedSchemeException($dsn, 'postal', $this->getSupportedSchemes());
+        }
+
+        $host = $dsn->getHost();
+        $port = $dsn->getPort();
+        $apiToken = $this->getPassword($dsn);
+
+        return (new PostalApiTransport($apiToken, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
```

Comment:
given that the host always must be set I would make it a mandatory argument (after `$apiToken`) for the constructor of `PostalApiTransport`) and validate here before that it is set in the DSN